tests/integration/master.test.js

```diff
@@ -0,0 +1,255 @@
+    let sto = await nc.rpc.state.getStorage(key);
+    assert(sto.isNone, "storage item should not have been set");
+  }, 20000);
+
+  test('Root call with valid votes and oversufficient vote count', async () => {
+    let key = u8a_hex(randomish_u8a());
+    let val = u8a_hex(randomish_u8a());
+    const did_to_key = [
+      [ALICE_DID, await keypair(ALICE_SK)],
+      [BOB_DID, await keypair(BOB_SK)],
+      [CHARLIE_DID, await keypair(CHARLIE_SK)],
+    ];
+    await master_set_storage(nc, key, val, did_to_key);
+    let sto = await nc.rpc.state.getStorage(key);
+    let u8a = sto.unwrap();
+    expect(u8a_hex(u8a)).toEqual(val);
+  }, 20000);
+
+  test('Root call with votes not sorted lexically', async () => {
+    let key = u8a_hex(randomish_u8a());
+    let val = u8a_hex(randomish_u8a());
+    const did_to_key = [
+      [BOB_DID, await keypair(BOB_SK)],
+      [ALICE_DID, await keypair(ALICE_SK)],
+    ];
+    await master_set_storage(nc, key, val, did_to_key);
+    let sto = await nc.rpc.state.getStorage(key);
+    let u8a = sto.unwrap();
+    expect(u8a_hex(u8a)).toEqual(val);
+  }, 20000);
+});
+
+// connect to running node
+async function connect() {
+  const extraTypes = {
+    Address: 'AccountId',
+    LookupSource: 'AccountId',
+  };
+  return await ApiPromise.create({
+    provider: new WsProvider(FullNodeEndpoint),
+    types: {
+      ...types,
+      ...extraTypes,
+    },
+  });
+}
+
+// load a DID kp from secret
+async function keypair(seed) {
+  assert(seed instanceof Uint8Array, "wrong type");
+  await cryptoWaitReady();
+  let keyring = new Keyring({ type: 'sr25519' });
+  let key = keyring.addFromSeed(seed);
+  return key
+}
+
+async function sign_send_tx(extrinsic) {
+  let key = await get_test_account_key();
+  await extrinsic.signAsync(key);
+
+  const promise = new Promise((resolve, reject) => {
+    try {
+      let unsubFunc = null;
+      return extrinsic.send(({ events = [], status }) => {
+        if (status.isFinalized) {
+          unsubFunc();
+          resolve({
+            events,
+            status,
+          });
+        }
+      })
+        .catch((error) => {
+          reject(error);
+        })
+        .then((unsub) => {
+          unsubFunc = unsub;
+        });
+    } catch (error) {
+      reject(error);
+    }
+
+    return this;
+  });
+
+  return await promise;
+}
+
+async function get_test_account_key() {
+  await cryptoWaitReady();
+  let keyring = new Keyring(TestKeyringOpts);
+  let key = keyring.addFromUri(TestAccountURI);
+  return key
+}
+
+// represent a Uint8Array as hex with a "0x" prefix
+function u8a_hex(bs) {
+  assert(bs instanceof Uint8Array, "wrong type");
+  return '0x' + [...bs].map(bt => ('0' + bt.toString(16)).slice(-2)).join('');
+}
+expect(u8a_hex(new Uint8Array([]))).toEqual('0x');
+expect(u8a_hex(new Uint8Array([0x01]))).toEqual('0x01');
+expect(u8a_hex(new Uint8Array([0xaa, 0xbb]))).toEqual('0xaabb');
+
+function randomish_u8a() {
```

Comment:
Whats the purpose this function? Are you looking for a pseudorandom 32 byte UintArray or with something slightly different? For the former, polkadot-js has `randomAsU8a`

Reply:
That's correct. removed `randomish_u8a` in favor of `randomAsU8a` in 974bafd.